src/cycod/Helpers/VisionHeaderPolicy.cs

```diff
@@ -57,11 +57,50 @@ private bool ContainsImageContent(PipelineMessage message)
             var contentData = BinaryData.FromStream(contentStream);
             var contentString = contentData.ToString();
 
-            if (!contentString.Contains("attached content")) return false;
-            if (!contentString.Contains("\"type\":\"image")) return false;
-            if (!contentString.Contains("\"url\":\"data:image")) return false;
+            // Parse the content string as JSON and check for image type and data URL
+            using var jsonDoc = JsonDocument.Parse(contentString);
+            bool hasImageType = false;
+            bool hasDataUrl = false;
 
-            return true;
+            void Traverse(JsonElement element)
```

Comment:
i typically don't use local functions, as it 'bloats' the main function... i typically would moe this to another private function in the class... also, doing an actual parse is very expensive, and in fact can fail in some edge caess i've seen... thus, the simple string searches... is that what was causing the problem you were seeing?